Add documentation for wp-env.json config file (#18643)
WordPress/gutenberg: packages/env/README.md

```

<br/><br/><p align="center"><img src="https://s.w.org/style/images/codeispoetry.png?1" alt="Code is Poetry." /></p>

> ## The Configuration File

Reviewer: ```suggestion
## Additional Configuration and Running with Multiple Plugins/Themes
```

> You may also specify a configuration file for `wp-env`. This can be useful for loading other themes and plugins that you are developing simultaneously. The script will attach the specified theme and plugin directories as volumes on the docker container so that changes you make to them exist in the WordPress instance.

Reviewer: ```suggestion
`wp-env` also supports a configuration file. At the moment, this is only used for loading extra themes and plugins that you may be developing together with your main one. The script will attach the specified theme and plugin directories as volumes on the docker containers so that changes you make to them exist in the WordPress instance.
```

### Example:

> The config file is plain JSON with the name `wp-env.json`. The format of the file looks like this:

Reviewer: ```suggestion
```

`wp-env.json`
```json
{
  "themes": [
    "../path/to/theme/dir"
  ],
  "plugins": [
    "../path/to/plugin/dir"
  ]
}
```

> ### Specifc Usage:

Reviewer: ```suggestion
### Caveats:
```

> This file should be located in the same directory from which you run `wp-env` commands for a project. So if you are running `wp-env` in the root directory of a plugin, `wp-env.json` should also be located there. 

Reviewer: ```suggestion
The file should be located in the same directory from which you run `wp-env` commands for a project. So if you are running `wp-env` in the root directory of a plugin, `wp-env.json` should also be located there. 
```

> Each item in the `themes` or `plugins` array should be an absolute or relative path to the root of a different theme or plugin directory. Relative paths will be resolved from the current working directory of the script, which means they are resolved from the location of the `wp-env.json` file.

Reviewer: ```suggestion
Each item in the `themes` or `plugins` array should be an absolute or relative path to the root of a different theme or plugin directory. Relative paths will be resolved from the current working directory, which means they will be resolved from the location of the `wp-env.json` file.
```
